sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java

```diff
@@ -304,4 +310,45 @@ public static void setLastCompletedUpdateType(
             return ConfigurationUpdater.UpdateResult.DeploymentType.valueOf(value);
         }
     }
+
+    /**
+     * Retrieves the contents of a file based on file name.
+     * @param stateStore The state store to get file content from.
+     * @param fileName The name of the file to retrieve.
+     * @return Contents of the file.
+     * @throws UnsupportedEncodingException
+     */
+    public static String getFile(StateStore stateStore, String fileName) throws UnsupportedEncodingException {
+        fileName = FILE_NAME_PREFIX + fileName;
+        return new String(stateStore.fetchProperty(fileName), FILE_ENCODING);
+    }
+
+    /**
+     * Stores the file in the state store.
+     * @param stateStore The state store to store the file in.
+     * @param fileName The name of the file to store.
+     * @param uploadedInputStream The input stream holding the content of the file.
+     */
+    public static void storeFile(StateStore stateStore, String fileName, InputStream uploadedInputStream)
+            throws StateStoreException, IOException {
+        StringWriter stringWriter = new StringWriter();
+        IOUtils.copy(uploadedInputStream, stringWriter, FILE_ENCODING);
```

Comment:
Need to limit the size of this copy. This check is needed in addition to any size check against the metadata above.
 Malicious input could have a `Content-Length` of 1024byte and an actual content length of infinity. E.g. piping /dev/zero into HTTP.

Reply:
Ah, in memory of heartbleed.